pkg/virt-api/webhooks/defaults.go

```diff
@@ -28,25 +28,104 @@ import (
 	v1 "kubevirt.io/api/core/v1"
 	"kubevirt.io/client-go/log"
 
+	apiinstancetype "kubevirt.io/api/instancetype"
+	instancetypev1beta1 "kubevirt.io/api/instancetype/v1beta1"
+
+	"kubevirt.io/kubevirt/pkg/instancetype"
 	"kubevirt.io/kubevirt/pkg/liveupdate/memory"
 	"kubevirt.io/kubevirt/pkg/network/vmispec"
 	"kubevirt.io/kubevirt/pkg/util"
 	virtconfig "kubevirt.io/kubevirt/pkg/virt-config"
 )
 
-func SetDefaultVirtualMachine(clusterConfig *virtconfig.ClusterConfig, vm *v1.VirtualMachine) error {
-	if err := setDefaultVirtualMachineInstanceSpec(clusterConfig, &vm.Spec.Template.Spec); err != nil {
-		return err
+func SetVirtualMachineDefaults(vm *v1.VirtualMachine, clusterConfig *virtconfig.ClusterConfig, instancetypeMethods instancetype.Methods) {
+	setDefaultInstancetypeKind(vm)
+	setDefaultPreferenceKind(vm)
+	preferenceSpec := getPreferenceSpec(vm, instancetypeMethods)
+	setDefaultArchitecture(clusterConfig, &vm.Spec.Template.Spec)
+	setVMDefaultMachineType(vm, preferenceSpec, clusterConfig)
+	setPreferenceStorageClassName(vm, preferenceSpec)
+}
+
+func getPreferenceSpec(vm *v1.VirtualMachine, instancetypeMethods instancetype.Methods) *instancetypev1beta1.VirtualMachinePreferenceSpec {
+	if vm.Spec.Preference == nil {
+		return nil
+	}
+
+	preferenceSpec, err := instancetypeMethods.FindPreferenceSpec(vm)
+	if err != nil {
+		// Log but ultimately swallow any preference lookup errors here and let the validating webhook handle them
+		log.Log.Reason(err).Error("Ignoring error attempting to lookup PreferredMachineType.")
+		return nil
+	}
```

Comment:
nit - Looking at this again I'd drop the error handling here.

Reply:
fair enough. I'll drop it.